commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/AbstractVfsContainer.java

```diff
@@ -81,11 +82,7 @@ public void close() {
         }
 
         // Close all components
-        for (final Object component : toclose) {
-            if (component instanceof VfsComponent) {
-                final VfsComponent vfsComponent = (VfsComponent) component;
-                vfsComponent.close();
-            }
-        }
+        Arrays.stream(toclose).filter(component -> component instanceof VfsComponent)
+                              .forEach(component -> ((VfsComponent) component).close());
```

Comment:
@PeterAlfredLee 
May you please rebase on `master`, test and cast to `java.io.Closeable` instead of `VfsComponent`.

Reply:
It seems the `close` in `java.io.Closeable` may throw an IOException while the `close` in `VfsComponent` won't. This means we have to surround this with a try-catch, which I think is not a good idea.

```
        // Close all components
        Arrays.stream(toclose).filter(component -> component instanceof VfsComponent)
                              .forEach(component -> {
                                  try {
                                      ((java.io.Closeable) component).close();
                                  } catch (IOException e) {
                                      e.printStackTrace();
                                  }
                              });
```

Reply:
You're right, no need for the extra layer.